qmlti/lib.php

```diff
@@ -511,13 +533,14 @@ function get_assessment_list_by_administrator($id) {
  *
  * returns TRUE if coaching report is valid
  */
-function is_coaching_report_available($db, $resource_link_id, $assessment_id, $participant_name) {
+function is_coaching_report_available($db, $consumer_key, $resource_link_id, $assessment_id, $participant_name) {
   $data_connector = LTI_Data_Connector::getDataConnector(TABLE_PREFIX, $db, DATA_CONNECTOR);
-   if ($data_connector->ReportConfig_loadAccessible($resource_link_id, $assessment_id) == 1) {
-      $result_id = get_result_id($participant_name);
-      if (( $result_id != FALSE ) && (count( (array)$result_id) != 0 )) {
+   if ($data_connector->ReportConfig_loadAccessible($consumer_key, $resource_link_id, $assessment_id)) {
+      if (get_result_id($participant_name) != FALSE) {
```

Comment:
you dont need to compare with FALSE flip the condition for readability:   if (get_result_id($participant_name))